idalib/src/idb.rs

```diff
@@ -300,6 +301,62 @@ impl IDB {
         Bookmarks::new(self)
     }
 
+    pub fn find_text(&self, start_ea: Address, text: impl AsRef<str>) -> Option<Address> {
+        let s = CString::new(text.as_ref()).ok()?;
+        let addr = unsafe { idalib_find_text(start_ea.into(), s.as_ptr()) };
+        if addr == BADADDR {
+            None
+        } else {
+            Some(addr.into())
+        }
+    }
+
+    pub fn find_text_all(&self, text: impl AsRef<str>) -> Option<Vec<Address>> {
```

Comment:
I'd switch this to return an iterator instead, e.g.:

```rust
    pub fn find_text_iter<'a, T>(&'a self, text: T) -> impl Iterator<Item = Address> + 'a
    where
        T: AsRef<str> + 'a,
    {
        let mut cur = 0u64;
        std::iter::from_fn(move || {
            let found = self.find_text(cur, text.as_ref())?;
            cur = self.find_defined(found).unwrap_or(BADADDR.into());
            Some(found)
        })
    }
```

We can then build any collection we like, e.g., `idb.find_text_iter("...").collect::<...>();`

Reply:
Ohh, I like this! Thank you for pointing it out, that's much better. It's sometimes hard for me to "think in Rust" after decades of C programming 😅